org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/TokenType.java

```diff
@@ -32,7 +33,17 @@
 	PROPERTY("property"),
 	VARIABLE("variable"),
 	PARAMETER("parameter"),
-	MODIFIER("modifier");
+	MODIFIER("modifier"),
+
+	// Java-specific token types. These should be prefixed as such,
+	// to avoid confusion with similarly named concepts in other langugages.
+	// For example, "module" typically has the same meaning as "namespace"
+	// in other languages (Rust, for example). But in Java, modules are a
+	// completely separate concept, packages being the "namespace" tokens.
+	MODULE("javaModule"),
```

Comment:
"module" is fine with me, as this is a Java language server.

Reply:
After thinking for a while, or we can directly re-use `namespace` for modules. I don't see any case in Java to distinguish packages and modules, as they don't occur in the same file...  both are some kind of `namespace`.

Reply:
> "module" is fine with me, as this is a Java language server.

Well, the problem is that the token types from the jdt.ls will coexist with token types from other language servers. As a theme author, you won't necessarily know what language you're styling the tokens for (that's why LSP defines "standard", generic token types). So someone who doesn't know about Java modules might see that the token type `module` is available, and style it based on the expectation that it has the same meaning as `namespace`. By prefixing it with `java`, it becomes clear that this is a Java-specific concept, and is semantically different from, let's say, modules in Rust. I don't think it matters *that much* in this particular case, but out of principle I made the decision to prefix it. Here's a quote from aeschli about this from [an earlier discussion](https://github.com/microsoft/vscode/issues/103097#issuecomment-672001256):
> If a custom token type or modifier is very language specific, it makes sense to prefix it that way: `javaMember` (extends member). But I don't think `javaMember` is a good example of a very 'java' specific concept.

This is also how the C# extension handles many language-specific token types, see [their `semanticTokenTypes` contribution](https://github.com/OmniSharp/omnisharp-vscode/blob/c09bf3f65319cc98fbf37f0070b9786ca4b16357/package.json#L3317-L3470). Additionally, [they even declare a `module` token type](https://github.com/OmniSharp/omnisharp-vscode/blob/c09bf3f65319cc98fbf37f0070b9786ca4b16357/package.json#L3415), which could potentially cause confusion with Java modules if they were not named `javaModule`. Although since the token type has no description I'm not entirely sure what it's used for anyway, especially since I'm no C# expert.

> After thinking for a while, or we can directly re-use `namespace` for modules. I don't see any case in Java to distinguish packages and modules, as they don't occur in the same file... both are some kind of `namespace`.

They *can* appear in the same file, since there are directives in module-info which refer to both module names and package names. Just see the test case for modules and you'll see what I mean. Personally I might style packages and modules the same, but they are semantically different and I don't like the idea of taking away customization abilities from theme authors without good reason.

If you want me to rename `javaModule` to `module` or just `namespace` I'm personally fine with that, but I think you should consider the above first.